packages/core/src/MachineNode.ts

```diff
@@ -63,7 +58,7 @@ export class MachineNode<
   TContext = any,
   TEvent extends EventObject = EventObject,
   TTypestate extends Typestate<TContext> = any
-> extends StateNode<TContext, TEvent> {
```

Comment:
so maybe it's not a node in the end and we should rename `MachineNode` to a `Machine`?

Reply:
Or `StateMachine` to prevent confusion (since v4 used `Machine` for something slightly different)

Reply:
That works too 👍

Reply:
Done in 9f554cd